qiita_ware/ebi.py

```diff
@@ -981,3 +950,52 @@ def send_xml(self):
             print "FAILED"
 
         return (study_accession, submission_accession)
+
+    def generate_demultiplexed_fastq(self):
+        """Generates demultiplexed fastq
+
+        Returns
+        -------
+        demux_samples
+            List of successful demultiplexed samples
+
+        Notes
+        -----
+        - As a performace feature, this method will check if self.ebi_dir
+        already exist and, if it does, the script will assume that in a
+        previous execution this step was performed correctly and will simply
+        read the file names from self.ebi_dir
+        """
+        ppd = self.preprocessed_data
+
+        if not isdir(self.ebi_dir):
+            makedirs(self.ebi_dir)
+
+            demux = [path for _, path, ftype in ppd.get_filepaths()
+                     if ftype == 'preprocessed_demux'][0]
+
+            demux_samples = set()
+            with open_file(demux) as demux_fh:
+                for s, i in to_per_sample_ascii(demux_fh,
+                                                self.sample_template.keys()):
+                    sample_fp = join(self.ebi_dir, "%s.fastq.gz" % s)
+                    wrote_sequences = False
+                    with gzopen(sample_fp, 'w') as fh:
+                        for record in i:
+                            fh.write(record)
+                            wrote_sequences = True
+
+                    if wrote_sequences:
+                        demux_samples.add(s)
+                    else:
+                        remove(sample_fp)
+
+        else:
+            demux_samples = set()
```

Comment:
Just to be clear what I mean is that you could do `for f in glob.glob(path.join(self.ebi_dir, '*.fastq.gz')`.